lib/ktsllex/topics.ex

```diff
@@ -46,27 +46,42 @@ defmodule Ktsllex.Topics do
   #     },
   #     "schemaRegistryDelete": true
   # }
+  #
+  # version 2.1
+  # "a1f44cb8-0f37-4b96-828c-57bbd8d4934b"
   defp get_token(host, user, password) do
     %{user: user, password: password}
     |> Poison.encode!()
     |> post(host <> @login_path)
     |> extract_body()
     |> decode()
-    |> extract_token()
   end
 
-  defp extract_body({:ok, %HTTPoison.Response{body: body}}), do: body
+  defp extract_body({:ok, %HTTPoison.Response{body: body, headers: headers}}) do
+    case gzipped(headers) do
+      true -> :zlib.gunzip(body)
+      false -> body
+    end
+  end
 
   defp extract_body({:error, %HTTPoison.Error{reason: reason}}) do
     Logger.error("#{__MODULE__} failed:#{inspect(reason)}")
     :error
   end
 
-  defp decode(:error), do: :error
-  defp decode(body), do: body |> Poison.decode!()
+  defp gzipped(headers) do
+    headers
+    |> Enum.any?(fn kv ->
```

Comment:
Feels like there should be a better way of doing this, but if it works can be refactored later!

Reply:
yeah, I was not keen, but might be solved much better soon https://github.com/edgurgel/httpoison/issues/81#issuecomment-153150985